components/engine/src/rocks/util/event_listener.rs

```diff
@@ -64,6 +66,26 @@ impl rocksdb::EventListener for EventListener {
             .inc();
     }
 
+    fn on_background_error(&self, reason: DBBackgroundErrorReason, result: Result<(), String>) {
+        assert!(result.is_err());
+        if let Err(err) = result {
+            let r = match reason {
+                DBBackgroundErrorReason::Flush => "flush",
+                DBBackgroundErrorReason::Compaction => "compaction",
+                DBBackgroundErrorReason::WriteCallback => "write_callback",
+                DBBackgroundErrorReason::MemTable => "memtable",
+            };
+            // Avoid tikv from restarting if rocksdb get corruption.
+            if err.starts_with("Corruption") {
+                set_panic_mark();
+            }
+            panic!(
```

Comment:
is it better to use fatal logging directly? seem we don't care about the panic backtrace?

/cc @BusyJay

Reply:
seems we do have logging before panic?
https://github.com/tikv/tikv/blob/master/components/tikv_util/src/lib.rs#L481

Reply:
panic is to restart tikv here, since after bg error rocksdb will reject all writes, where tikv will panic anyway (for not able to write).